weblate/trans/tasks.py

```diff
@@ -67,10 +67,9 @@ def perform_update(cls, pk, auto=False):
             obj = Project.objects.get(pk=pk)
         else:
             obj = Component.objects.get(pk=pk)
-
-        if not auto or settings.AUTO_UPDATE:
+        if settings.AUTO_UPDATE in ("full", True) or auto:
             obj.do_update()
-        else:
+        elif settings.AUTO_UPDATE in ("remote", False):
```

Comment:
Using just `else` should be enough here, other values are handled in update_remotes.